### PR TITLE
Prometheus stuff

### PR DESCRIPTION
**Pull request by coolreader18**
# Description of Changes

Sorry, still in a bit of a funk, but figured I may as well just push what I have.

# API and ABI

 - [ ] This is a breaking change to the module ABI
 - [ ] This is a breaking change to the module API
 - [ ] This is a breaking change to the ClientAPI
 - [ ] This is a breaking change to the SDK API

*If the API is breaking, please state below what will break*

### CODE REVIEW

File: crates/core/src/database_logger.rs [hunk 1]
```diff
@@ -125,7 +125,7 @@ impl DatabaseLogger {
 
     pub fn filepath(address: &Address, instance_id: u64) -> PathBuf {
         let root = crate::stdb_path("worker_node/database_instances");
-        root.join(address.to_hex())
+        root.join(&*address.to_hex())
```
Comment (kim) [inline crates/core/src/database_logger.rs:128]: What's the significance of these reborrows (here and in database_instance_context.rs)?

File: crates/core/src/json/client_api.rs [hunk 2]
```diff
@@ -78,8 +79,8 @@ pub struct SubscriptionUpdateJson {
 #[derive(Debug, Clone, Serialize)]
 pub struct EventJson {
     pub timestamp: u64,
-    pub status: String,          // committed, failed
-    pub caller_identity: String, // hex identity
+    pub status: String,            // committed, failed
+    pub caller_identity: Identity, // hex identity
```
Comment (kim) [inline crates/core/src/json/client_api.rs:83]: `Identity` is not serialized through hex afaics, so this is essentially breaking. Not sure what consumes this value, though.

Is this change necessary for what this patch is trying to achieve?

File: crates/core/src/auth/identity.rs [hunk 3]
```diff
@@ -39,7 +39,7 @@ pub fn encode_token_with_expiry(
     });
 
     let claims = SpacetimeIdentityClaims {
-        hex_identity: identity.to_hex(),
+        hex_identity: identity.to_hex().to_string(),
```
Comment (kim) [inline crates/core/src/auth/identity.rs:42]: It's a string already, no?

**Review by kim — COMMENTED** [review]
This seems like an improvement overall, but I'd appreciate a bit of tidying.

We'll also need a follow-up to integrate into cloud.

File: crates/core/src/worker_metrics/mod.rs [hunk 4]
```diff
@@ -1,213 +1,210 @@
...
-    // instance_env_delete_value: HistogramVec,
-    instance_env_delete_eq: HistogramVec,
-    // instance_env_delete_range: HistogramVec,
-}
+use prometheus::core::{Metric, MetricVec, MetricVecBuilder};
+use prometheus::{Gauge, GaugeVec, HistogramVec, IntCounterVec, IntGauge, IntGaugeVec};
+use spacetimedb_lib::{Address, Hash, Identity};
+
+#[macro_export]
+macro_rules! metrics_group {
```
[the remark this answers is not in the document]
Comment (coolreader18) [reply crates/core/src/worker_metrics/mod.rs:7]: Oh wow, there definitely is. That's funny lol, we had almost the same idea. I think in the future I'll look at expanding it to something more general (unsure about pulling it out into its own crate that I then publish or something, wrt licensing) but the one on crates.io doesn't let you make fields public (??) or else I might consider switching. I'm happy with the strong typing with this implementation, as well (not just strings everywhere).

File: crates/core/src/database_logger.rs (hunk 1 above)
Comment (coolreader18) [reply crates/core/src/database_logger.rs:128]: Oh, yes. There was another part of this that I ended up splitting off that changed stuff around Address and Identity hex stuff.

File: crates/core/src/worker_metrics/mod.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (kim) [reply crates/core/src/worker_metrics/mod.rs:7]: I meant maybe a workspace crate, or in lib, or somewhere near the top of the core crate module hierarchy. Just so me and others can find it easily for instrumenting other stuff.

[the remark this answers is not in the document]
Comment (coolreader18) [reply crates/core/src/worker_metrics/mod.rs]: Ok, moved it to a submodule of `spacetimedb_core::util`

**Comment by drogus** [comment]
I just wanted to say that I've been running with this changes on dev for a week or so and observing some of the metrics (like the queue length) and it works without any issues. Thanks for great work Noa!

**Review by drogus — APPROVED** [review]
Great work!
